Examples/Io/Root/src/RootDigitizationWriter.cpp

```diff
@@ -0,0 +1,165 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2017 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "ActsExamples/Io/Root/RootDigitizationWriter.hpp"
+#include "ActsExamples/EventData/SimHit.hpp"
+#include "ActsExamples/EventData/SimIdentifier.hpp"
+#include "ActsExamples/EventData/SimParticle.hpp"
+#include "ActsExamples/Framework/WhiteBoard.hpp"
+#include "ActsExamples/Utilities/Paths.hpp"
+#include <Acts/Surfaces/Surface.hpp>
+#include <Acts/Utilities/Helpers.hpp>
+#include <Acts/Utilities/Intersection.hpp>
+#include <Acts/Utilities/Units.hpp>
+
+#include <ios>
+#include <optional>
+#include <stdexcept>
+
+#include <TFile.h>
+#include <TString.h>
+
+ActsExamples::RootDigitizationWriter::RootDigitizationWriter(
+    const ActsExamples::RootDigitizationWriter::Config& cfg,
+    Acts::Logging::Level lvl)
+    : WriterT(cfg.inputMeasurements, "RootDigitizationWriter", lvl),
+      m_cfg(cfg),
+      m_outputFile(cfg.rootFile) {
+  // inputClusters is already checked by base constructor
+  if (m_cfg.inputSimulatedHits.empty()) {
+    throw std::invalid_argument("Missing simulated hits input collection");
+  }
+  // Setup ROOT I/O
+  if (m_outputFile == nullptr) {
+    m_outputFile = TFile::Open(m_cfg.filePath.c_str(), m_cfg.fileMode.c_str());
+    if (m_outputFile == nullptr) {
+      throw std::ios_base::failure("Could not open '" + m_cfg.filePath);
+    }
+  }
+  m_outputFile->cd();
+
+  // Analyze the smearers
+  if (not m_cfg.smearers.empty()) {
+    ACTS_DEBUG("Smearers are present, preparing trees.");
+    for (auto& [key, value] : m_cfg.smearers) {
+      auto& geoID = key;
+      std::visit(
+          [&](auto&& smearer) {
+            auto dTree = std::make_unique<DigitizationTree>(geoID);
```

Comment:
Are we having a tree for each individual surface here? Can't we write into just one tree?

Reply:
No, it's actually per smearer (in a volume), with the next commit this will be clearer. 
For each individual smearer with its unique smarting configuration there will be a dedicated tree. If you smear all hits in one volume the same, you will only have one tree, if you decide to have a different resolution per layer you will have one tree per layer.

Reply:
The `GeometryHierarchyMap` will do its job here: it gives the closest in hierarchy back.